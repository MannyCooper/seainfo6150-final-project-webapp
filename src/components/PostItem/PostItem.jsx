import React from 'react';
import { Link } from "react-router-dom";
import styles from "./PostItem.module.css"
import moment from "moment";

const PostItem = (props) => {
    return(
        <li className={styles.post_item}>
                <header className={styles.header}>
                    <img className={styles.image} src={props.post.Cover} alt={props.post.Title}/>
                </header>
                <div className={styles.description}>
                    <article>
                        <Link className={styles.category} to="/">{props.post.Category}</Link>
                        <h2 className={styles.post_title}>{props.post.Title}</h2>
                    </article>
                    <time className={styles.date} dateTime={props.post.Date}>
                        {moment(props.post.Date, 'YYYY-MM-DD').format('MMM Do, YYYY')}
                    </time>
                </div>
                <Link className={styles.post_link} to={`/post/${props.post.Slug}`} />
        </li>
    )
}

export default PostItem;