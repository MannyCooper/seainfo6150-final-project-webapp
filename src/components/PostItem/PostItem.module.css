@import url('https://fonts.googleapis.com/css2?family=Open+Sans:wght@700&display=swap');

.post_item {
    background-color: white;
    border-radius: 25px;
    overflow: hidden;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    flex-basis: auto;
    padding: 32px;
    position: relative;
    transition: transform 0.3s, box-shadow 0.6s;
    width: 346px;
    padding: 26px;
    margin-bottom: 36px;
    margin-right: 30px;
}

.post_item:nth-child(3n){
    margin-right: 0;
}

.post_item:hover{
    transform: translateY(-5px);
    box-shadow: 0 5px 30px 0 rgba(0,0,0,0.05);
}

.post_item:hover .image{
    -webkit-transform: scale(1.03);
    transform: scale(1.03);
}


.header{
    margin: -32px;
    margin-bottom: 16px;
}

.image{
    width: 100%;
    height: 100%;
    object-fit: cover;
    transition: transform 0.3s;
}

.description{
    justify-content: space-between;
    flex-basis: auto;
    display: flex;
    flex-direction: column;
    flex-shrink: 0;
    flex-grow: 1;
}

.category{
    margin-bottom: 8px;
    font-size: 12px;
    line-height: 1.3;
    font-weight: 700;
    color: #6e6e73;
    letter-spacing: 0em;
}

.post_title{
    font-family: 'Open Sans', sans-serif;
    font-size: 24px;
    line-height: 1.2;
    font-weight: 700;
    letter-spacing: 0.009em;
    max-height: 7em;
}

.date{
    margin-top: 12px;
    font-size: 13.5px;
    line-height: 1.3;
    font-weight: 500;
    letter-spacing: 0.02em;
    color: #6e6e73;
}

.post_link{
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    opacity: 0;
}

@media only screen and (max-width: 1380px){
    .post_item{
        width: 472px;
        padding: 32px;
        margin-bottom: 36px;
    }
    .post_item:nth-child(3n){
        margin-right: 0px;
    }
    .post_item:nth-child(even){
        margin-right: 0px;
    }
    .post_item:nth-child(odd){
        margin-right: 36px;
    }
    .header{
        height: 266px;
    }
}

@media only screen and (max-width: 1200px){
    .post_item{
        width: 333px;
        padding: 24px;
        margin-bottom: 24px;
    }
    .post_item:nth-child(odd){
        margin-right: 24px;
    }
    .header{
        height: 187px;
    }
}

@media only screen and (max-width: 768px){
    .post_item{
        width: auto;
        padding: 24px;
        margin-bottom: 24px;
        margin-left: auto;
        margin-right: auto;
    }
    .post_item:nth-child(odd){
        margin-right: auto;
    }
    .header{
        height: 187px;
    }
}