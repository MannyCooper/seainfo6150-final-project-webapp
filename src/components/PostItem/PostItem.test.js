 /* global render */
 import React from "react";
 import PostItem from "./PostItem";
 
 const test = {
    "ux-designer": {
        "Slug": "ux-designer",
        "Title": "7 UX Deliverables: What will I be making as a UX designer?",
        "Cover": "https://cdn.dribbble.com/assets/art-banners/romainbriaux-284c5096f64764a70b44faa03268b263c4b37365e988446318005ff367ced170.webp",
        "Content": "<p>Test.</p>",
        "Date": "2019-08-25",
        "Category": "Designer",
        "Tags": [
            "UX",
            "designer"
        ]
    }
 }

 it("renders the ArchiveItem component correctly", () => {
   const { container } = render(<PostItem post={test} key={test.Slug}/>);
   expect(container).toMatchSnapshot();
 });
 