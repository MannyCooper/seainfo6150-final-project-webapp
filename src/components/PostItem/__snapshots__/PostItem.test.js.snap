// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ArchiveItem component correctly 1`] = `
<div>
  <li
    class="post_item"
  >
    <header
      class="header"
    >
      <img
        class="image"
      />
    </header>
    <div
      class="description"
    >
      <article>
        <a
          class="category"
          href="/"
        />
        <h2
          class="post_title"
        />
      </article>
      <time
        class="date"
      >
        Invalid date
      </time>
    </div>
    <a
      class="post_link"
      href="/post/undefined"
    />
  </li>
</div>
`;
